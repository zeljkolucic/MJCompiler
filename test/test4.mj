program test5
	char c1, c2, c3;
	int a, b, c;
	int niz[], niz2[];
{
	void main()
	{
		read(c1);
		print(c1);
		
		read(a);
		print(a);
		
		read(b);
		print(b*2, 3);
		read(c);
		print(c, 10);
		
		do{
			read(a);
			print(a);
			c--;
		} while(c>0);	
	}
}