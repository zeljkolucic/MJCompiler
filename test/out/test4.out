Linija [6]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [7]: Pristup globalnoj promenljivoj x.
DesignatorIdent(
 x
) [DesignatorIdent]
Linija [9]: Pristup globalnoj promenljivoj array.
DesignatorIdent(
 array
) [DesignatorIdent]
Linija [10]: Pristup globalnoj promenljivoj array.
DesignatorIdent(
 array
) [DesignatorIdent]
Linija [10]: Pristup elementu niza array.
DesignatorIdentArray(
  DesignatorIdent(
   array
  ) [DesignatorIdent]
  PosTermExpr(
    Term(
      ConstFactor(
        NumConst(
         0
        ) [NumConst]
      ) [ConstFactor]
      NoMulopFactor(
      ) [NoMulopFactor]
    ) [Term]
  ) [PosTermExpr]
) [DesignatorIdentArray]
Linija [11]: Pristup globalnoj promenljivoj array.
DesignatorIdent(
 array
) [DesignatorIdent]
Linija [11]: Pristup elementu niza array.
DesignatorIdentArray(
  DesignatorIdent(
   array
  ) [DesignatorIdent]
  PosTermExpr(
    Term(
      ConstFactor(
        NumConst(
         0
        ) [NumConst]
      ) [ConstFactor]
      NoMulopFactor(
      ) [NoMulopFactor]
    ) [Term]
  ) [PosTermExpr]
) [DesignatorIdentArray]
Linija [13]: Pristup globalnoj promenljivoj y.
DesignatorIdent(
 y
) [DesignatorIdent]
Linija [14]: Pristup globalnoj promenljivoj y.
DesignatorIdent(
 y
) [DesignatorIdent]
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: , -1, -1 
Prog MJProgram: notype, 0, 1 
   Var x: int, 0, 0 
   Var y: char, 1, 0 
   Var array: Arr of int, 2, 0 
   Meth main: notype, 0, 0 



